### DISCUSSION TITLE
Best way to send to send HTTP Keep-Alive in RemoteGraphQLDataSource

### DISCUSSION THREAD

**Discussion by ehardy** [apollographql/apollo-server#5357] [Q&A] [answered]
Hello,

Looking for some advice how to best send the HTTP `Connection: Keep-Alive` header when using `RemoteGraphQLDataSource` in a gateway context. `RemoteGraphQLDataSource` has a `willSendRequest()` method that can be overridden, I could set the header in there. Deep down, Apollo gateway uses `node-fetch`, which itself uses the `http` and `https` Node modules, would there be a way to customize the Agent with `keepAlive` as can be seen in Node's documentation?

Thanks

Etienne

**Comment by glasser** [comment, marked as answer]
You can [configure the subgraph fetcher](https://www.apollographql.com/docs/federation/api/apollo-gateway/#configuring-the-subgraph-fetcher) in your GraphQLDataSource to replace the default `fetch` call with your own `fetch` call.  You can have your fetcher call `node-fetch` passing in an `agent`, for example.

**Reply by ehardy** [reply]
Looks promising! We're using an old version of `@apollo/gateway` and had not noticed this option, so will upgrade and try it out. Thanks.

**Reply by ehardy** [reply]
@glasser Confirm that your proposal works. Thanks!
